app/components/ui/BrowserCompatibilityModal.tsx

```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+import { toast } from 'sonner';
+import { classNames } from '~/utils/classNames';
+
+interface BrowserCompatibilityModalProps {
+  isOpen: boolean;
+}
+
+export const BrowserCompatibilityModal: React.FC<BrowserCompatibilityModalProps> = ({ isOpen }) => {
+  if (!isOpen) {
+    return null;
+  }
+
+  const handleCopyUrl = () => {
+    const currentUrl = window.location.href;
+
+    navigator.clipboard
+      .writeText(currentUrl)
+      .then(() => {
+        toast.success('URL copied to clipboard! Paste it in a supported browser for full functionality.');
+      })
+      .catch(() => {
+        const userInput = prompt(
+          'Copy this URL to open in a supported browser (Chrome, Edge, Brave, Opera):',
+          currentUrl,
+        );
+
+        if (userInput !== null) {
+          toast.success('Please open the URL in a supported browser for full functionality.');
+        }
+      });
+  };
+
+  return (
+    <div
+      className={classNames('fixed inset-0 flex items-center justify-center')}
+      style={{
+        backdropFilter: 'blur(8px)',
+        backgroundColor: 'rgba(0, 0, 0, 0.5)',
+        zIndex: 999999,
+      }}
+    >
+      <div
+        className={classNames('relative max-w-lg mx-4 p-10 rounded-2xl shadow-2xl', 'bg-depth-1 border')}
+        style={{
+          backgroundColor: 'rgb(var(--color-depth-1))',
+          borderColor: 'rgb(var(--color-accent-700))',
+          boxShadow: '0 25px 50px -12px rgba(0, 0, 0, 0.8), 0 0 0 1px rgba(59, 206, 255, 0.2)',
+        }}
+      >
+        {/* Warning Icon */}
+        <div className="flex justify-center mb-8">
+          <div
+            className="p-4 rounded-full relative"
+            style={{
+              backgroundColor: 'rgb(var(--color-accent-500))',
+              boxShadow: '0 0 30px rgba(59, 206, 255, 0.4)',
+            }}
+          >
+            <svg
+              className="w-12 h-12"
+              fill="currentColor"
+              viewBox="0 0 20 20"
+              style={{ color: 'rgb(var(--color-depth-1))' }}
+            >
+              <path
+                fillRule="evenodd"
+                d="M18 10a8 8 0 11-16 0 8 8 0 0116 0zm-7-4a1 1 0 11-2 0 1 1 0 012 0zM9 9a1 1 0 000 2v3a1 1 0 001 1h1a1 1 0 100-2v-3a1 1 0 00-1-1H9z"
+                clipRule="evenodd"
+              />
+            </svg>
+          </div>
+        </div>
+
+        {/* Title */}
+        <h2 className="text-3xl font-bold text-center mb-6" style={{ color: 'rgb(var(--color-primary))' }}>
```

Comment:
You can just use text-primary class 
```suggestion
        <h2 className="text-3xl font-bold text-center text-primary mb-6">
```